自分の環境で試した結果

`.c`と`*.o` -> 全ての.cファイルに、対となる`.o`ができる。

```
$ make
cc    -c -o main.o main.c
cc    -c -o command/command.o command/command.c
中略
cc    -c -o type/string_type.o type/string_type.c
cc    -c -o type/unsigned_type.o type/unsigned_type.c
gcc -g -Wall -Wextra -o main main.o command/command.o command/command_parse.o command/command_perform.o console/color.o console/screen.o file/reader.o file/writer.o keyboard/keyboard.o render/render.o render/render_body.o render/render_footer.o render/render_header.o render/render_util.o type/context_type.o type/cursor_type.o type/mbchar_type.o type/string_type.o type/unsigned_type.o
```

理由不明ですがccでビルドされている

---

`.c.o` -> ビルドエラーになる

```
$ make all
gcc -g -c main.o main.c
clang: error: no such file or directory: 'main.o'
make: *** [main.o] Error 1
```

なんか途中のメッセージが違うと思って、

```
.c.o:
	cc -c -o $@ $<
```

```
.c.o:
	gcc -c -o $@ $<
```

を試すと両方ビルドが通る($^でも通る)

---

```
%.o:
	gcc -c -o $@ $<
```
色々試しても%だと通らない
(他の人のMakefileの指定に%.oがあったので試してみた)

---

なお、環境は以下です。
```
$ make -v
GNU Make 3.81
Copyright (C) 2006  Free Software Foundation, Inc.
This is free software; see the source for copying conditions.
There is NO warranty; not even for MERCHANTABILITY or FITNESS FOR A
PARTICULAR PURPOSE.

This program built for i386-apple-darwin11.3.0

$ cc -v
Apple LLVM version 8.0.0 (clang-800.0.42.1)
Target: x86_64-apple-darwin16.1.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin

$ gcc -v
Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.12.sdk/usr/include/c++/4.2.1
Apple LLVM version 8.0.0 (clang-800.0.42.1)
Target: x86_64-apple-darwin16.1.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```

--- a/src/Makefile
+++ b/src/Makefile
@@ -9,7 +9,7 @@ $(PROG): $(OBJS)
 	gcc -g -Wall -Wextra -o $@ $^
 
 .c.o:
-	gcc -g -Wall -Wextra -c -o $@ $<
+	gcc -Wall -Wextra -c -o $@ $<
 
 clean:
 	rm -rf $(PROG) $(OBJS)

--- a/src/Makefile
+++ b/src/Makefile
@@ -9,7 +9,7 @@ $(PROG): $(OBJS)
 	gcc -g -Wall -Wextra -o $@ $^
 
 .c.o:
-	gcc -Wall -Wextra -c -o $@ $<
+	gcc -g -Wall -Wextra -c -o $@ $<
 
 clean:
 	rm -rf $(PROG) $(OBJS)

